js/theme/handlers.js

```diff
@@ -19,6 +19,20 @@ var buttons = {
         .then(function() {
           view.displayNewPosts();
         })
+    },
+    "[data-js-orderby-oldest-first]": function() {
+      viewmodel.loadPosts({sort: 'oldest_first', returnPromise: true})
+        .then(view.renderTimeline)
+        .then(function() {
+          view.toggleSortBtn('oldest_first')
+        })
+    },
+    "[data-js-orderby-newest-first]": function() {
+      viewmodel.loadPosts({sort: 'newest_first', returnPromise: true})
```

Comment:
As we discussed, the returnPromise conditional is a workaround. I will see what's going on and propose we not introduce that logic here.